lib/matplotlib/widgets.py

```diff
@@ -389,6 +370,29 @@ def __init__(self, ax, label, valmin, valmax, valinit=0.5, valfmt='%1.2f',
         self.slidermin = slidermin
         self.slidermax = slidermax
         self.drag_active = False
+        self._value_in_bounds(valinit)
+
+    def _value_in_bounds(self, val):
+        """ Makes sure self.val is with given bounds."""
+        if val <= self.valmin:
+            if not self.closedmin:
+                return
+            val = self.valmin
+        elif val >= self.valmax:
+            if not self.closedmax:
+                return
+            val = self.valmax
+
+        if self.slidermin is not None and val <= self.slidermin.val:
+            if not self.closedmin:
+                return
+            val = self.slidermin.val
+
+        if self.slidermax is not None and val >= self.slidermax.val:
+            if not self.closedmax:
+                return
+            val = self.slidermax.val
+        self.set_val(val)
```

Comment:
I think I would rather that this method returns the value. It can then be up to the caller what to do with it. Might make it more portable with the planned traitlets work.